packages/x-components/src/components/base-slider.vue

```diff
@@ -0,0 +1,174 @@
+          </span>
+        </p>
+        <p class="x-base-slider-selected-max">
+          <span>max value</span>
+          <span>
+            {{ rangeSelected[1] }}
+          </span>
+        </p>
+      </slot>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+  import { API, create } from 'nouislider';
+  import { computed, defineComponent, onMounted, onUnmounted, PropType, ref, watch } from 'vue';
+
+  export default defineComponent({
+    name: 'BaseSlider',
+    components: {},
+    props: {
+      threshold: {
+        type: Object as PropType<{ min: number; max: number }>,
+        default: () => ({ min: 0, max: Number.MAX_SAFE_INTEGER })
+      },
+      modelValue: {
+        type: Object as PropType<{ min: number; max: number }>,
+        required: true
+      }
+    },
+    emits: ['update:modelValue'],
+    setup(props, { emit }) {
+      let sliderInstance: API;
+      const slider = ref<HTMLElement>();
+
+      const minSelected = ref(props.modelValue?.min ?? props.threshold.min);
+      const maxSelected = ref(props.modelValue?.max ?? props.threshold.max);
+
+      const rangeSelected = computed(() => [minSelected.value, maxSelected.value]);
+      const slideRange = computed(() => ({ min: props.threshold.min, max: props.threshold.max }));
+
+      onMounted(() => {
+        sliderInstance = create(slider.value!, {
+          start: rangeSelected.value,
+          range: slideRange.value,
+          step: 1,
+          connect: true,
+          margin: 1
+        });
+
+        sliderInstance.on('update', ([min, max]) => {
+          minSelected.value = Number(min);
+          maxSelected.value = Number(max);
+        });
+
+        sliderInstance.on('change', () =>
+          emit('update:modelValue', { min: minSelected.value, max: maxSelected.value })
+        );
+      });
+
+      onUnmounted(() => {
+        // Waiting to finish the collapse animation before destroying it
+        setTimeout(sliderInstance.destroy.bind(sliderInstance), 600);
+      });
+
+      watch(
+        () => props.threshold,
+        ({ min, max }) => {
+          sliderInstance.updateOptions({ range: slideRange.value, start: [min, max] }, false);
+          emit('update:modelValue', { min, max });
+        }
+      );
+
+      watch([() => props.modelValue.min, () => props.modelValue.max], ([min, max]) => {
+        if (min === minSelected.value && max === maxSelected.value) {
+          return;
+        }
+
+        const minValidated = min < props.threshold.min ? props.threshold.min : min;
+        const maxValidated = max > props.threshold.max ? props.threshold.max : max;
+        sliderInstance.set([minValidated, maxValidated]);
+
+        if (minValidated !== min || maxValidated !== max) {
+          emit('update:modelValue', { min: minValidated, max: maxValidated });
+        }
+      });
+
+      return {
+        slider,
+        rangeSelected
+      };
+    }
+  });
+</script>
+
+<style lang="css">
```

Comment:
maybe scoped?

Reply:
No, it needs to be not scoped, so `nouislider` styles import works as needed.

Reply:
It needs to be not scoped, so the styles are correctly applied to the slider.